woocommerce/woocommerce-gateway-stripe
# Fix payment methods count on settings page

## Changes proposed in this Pull Request:
The condition [here](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/includes/class-wc-gateway-stripe.php#L145) is exclusive for legacy checkout but we were not checking if legacy is enabled or not. So this block of code runs for UPE as well making the payment methods count wrong. 

## Testing instructions
- Enable new checkout experience (UPE)
- Enable a few payment methods.
- Go to **WooCommerce > Settings > Payments** page.
- In `develop` branch, notice that the payment method count is always 1 beside the Stripe title.

<img width="1256" alt="Screenshot 2024-10-18 at 8 45 59 PM" src="https://github.com/user-attachments/assets/d0cfa504-dd5f-43d0-9da8-ab7aa6ac7cc0">

- In this branch you should see the correct number.
<img width="1256" alt="Screenshot 2024-10-18 at 8 37 08 PM" src="https://github.com/user-attachments/assets/3c580909-c4da-4324-903e-9184fc7d5dd3">
